### PR TITLE
Some date overrides ignored in availability check

### PR DESCRIPTION
**Pull request by mischarouleaux**
## What does this PR do?

This is a fix for the following issue: https://github.com/calcom/cal.com/issues/6622

When an user has date overrides, these will be taken into consideration when someone books a meeting.

Fixes # (6622)

Result without the code fix:
Screenshot 2023-02-06 at 15 12 19

Result with the code fix:
Screenshot 2023-02-06 at 15 12 36

**Environment**: Main branch

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Test A: Create a date override, and place a new booking within that timespan.
- [ ] Test B: Do the same, but now a timespan before the date override.
- [ ] Test C: Do the same, but now a timespan after the date override.
- [ ] Test D: No days available and no date overrides, user shouldn't be able to book the meeting
- [ ] Test E: Book a normal meeting with no date overrides.

## Checklist
- I haven't checked if my PR needs changes to the documentation

/claim #6622

### CODE REVIEW

**Comment by JeroenReumkens** [comment]
Code wise this pr looks like a great and simple improvement. Not sure if this touches all use cases though -- perhaps best if you could test on the preview url @PeerRich since you know exactly where the problem occurs?

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -158,6 +158,48 @@ const isWithinAvailableHours = (
...
+  }
+) => {
+  const timeSlotStart = dayjs(timeSlot.start).utc();
+  const timeSlotEnd = dayjs(timeSlot.end).utc();
+
+  for (const dateOverride of dateOverrides) {
+    const startTime = timeSlotStart
+      .startOf("day")
+      .add(dateOverride.start.getUTCHours(), "hour")
+      .add(dateOverride.start.getUTCMinutes(), "minute");
```
Comment (zomars) [inline packages/features/bookings/lib/handleNewBooking.ts:185]: Will this be affected by the DST bug? cc @roae to investigate 🙏🏽

**Review by zomars — CHANGES_REQUESTED** [review]
LGTM. But blocking to check for missing edge-cases.

File: packages/features/bookings/lib/handleNewBooking.ts (hunk 1 above)
Comment (roae) [reply packages/features/bookings/lib/handleNewBooking.ts:185]: Yes, here we must always keep in mind the time zone of the event/organizer and the booker.

We need to merge main in this branch to bring those fixes and apply them in this function. I will do it

**Comment by roae** [comment]
@PeerRich @zomars this pr is not redundant with this one? #6538

**Comment by PeerRich** [comment]
> @PeerRich @zomars this pr is not redundant with this one? #6538

@emrysal probably knows more

**Comment by zomars** [comment]
Closing as this seems fixed in https://github.com/calcom/cal.com/pull/7792

Feel free to reopen if needed.

Thank you again for your contribution 🙏

**Pull request closed without merging**